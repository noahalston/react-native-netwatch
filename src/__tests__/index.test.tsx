import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { IProps, Netwatch} from '../index';

describe('Index test suite', () => {
  let component: ShallowWrapper;
  let props: IProps;

  it('should render properly', () => {
    givenProps();
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  it('should render properly visible and enabled', () => {
    givenProps(true, true);
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  it('should render properly visible, enabled and maxRequest', () => {
    givenProps(true, true, 20);
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  // GIVEN
  function givenComponent() {
    component = shallow(<Netwatch {...props} />);
  }

  function givenProps(visible: boolean = false, enabled: boolean = false, maxRequests?: number) {
    props = {
      visible,
      enabled,
      onPressClose: jest.fn(),
      maxRequests,
    };
  }
});
