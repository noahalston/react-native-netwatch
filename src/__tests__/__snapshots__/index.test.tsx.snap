// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index test suite should render properly 1`] = `
<Provider>
  <ContextProvider
    value={
      Object {
        "black": "#000",
        "blue300": "#67E8F9",
        "blue500": "#06B6D4",
        "blue700": "#0E7490",
        "gray100": "#F3F4F6",
        "gray200": "#E5E7EB",
        "gray300": "#D1D5DB",
        "gray400": "#9CA3AF",
        "gray50": "#F9FAFB",
        "gray500": "#6B7280",
        "gray600": "#4B5563",
        "gray700": "#374151",
        "gray800": "#1F2937",
        "gray900": "#111827",
        "green700": "#4D7C0F",
        "orange700": "#D97706",
        "red700": "#B91C1C",
        "violet700": "#764ABC",
        "white": "#FFF",
      }
    }
  >
    <ForwardRef(SafeAreaView)>
      <Component
        animationType="slide"
        hardwareAccelerated={false}
        visible={false}
      >
        <Main
          clearAll={[Function]}
          maxRequests={50}
          nRequests={Array []}
          onPress={[Function]}
          onPressClose={[MockFunction]}
          onPressDetail={[Function]}
          reduxActions={Array []}
          rnRequests={Array []}
          testId="mainScreen"
          visible={false}
        />
      </Component>
    </ForwardRef(SafeAreaView)>
  </ContextProvider>
</Provider>
`;

exports[`Index test suite should render properly visible and enabled 1`] = `
<Provider>
  <ContextProvider
    value={
      Object {
        "black": "#000",
        "blue300": "#67E8F9",
        "blue500": "#06B6D4",
        "blue700": "#0E7490",
        "gray100": "#F3F4F6",
        "gray200": "#E5E7EB",
        "gray300": "#D1D5DB",
        "gray400": "#9CA3AF",
        "gray50": "#F9FAFB",
        "gray500": "#6B7280",
        "gray600": "#4B5563",
        "gray700": "#374151",
        "gray800": "#1F2937",
        "gray900": "#111827",
        "green700": "#4D7C0F",
        "orange700": "#D97706",
        "red700": "#B91C1C",
        "violet700": "#764ABC",
        "white": "#FFF",
      }
    }
  >
    <ForwardRef(SafeAreaView)>
      <Component
        animationType="slide"
        hardwareAccelerated={false}
        visible={true}
      >
        <Main
          clearAll={[Function]}
          maxRequests={50}
          nRequests={Array []}
          onPress={[Function]}
          onPressClose={[MockFunction]}
          onPressDetail={[Function]}
          reduxActions={Array []}
          rnRequests={Array []}
          testId="mainScreen"
          visible={true}
        />
      </Component>
    </ForwardRef(SafeAreaView)>
  </ContextProvider>
</Provider>
`;

exports[`Index test suite should render properly visible, enabled and maxRequest 1`] = `
<Provider>
  <ContextProvider
    value={
      Object {
        "black": "#000",
        "blue300": "#67E8F9",
        "blue500": "#06B6D4",
        "blue700": "#0E7490",
        "gray100": "#F3F4F6",
        "gray200": "#E5E7EB",
        "gray300": "#D1D5DB",
        "gray400": "#9CA3AF",
        "gray50": "#F9FAFB",
        "gray500": "#6B7280",
        "gray600": "#4B5563",
        "gray700": "#374151",
        "gray800": "#1F2937",
        "gray900": "#111827",
        "green700": "#4D7C0F",
        "orange700": "#D97706",
        "red700": "#B91C1C",
        "violet700": "#764ABC",
        "white": "#FFF",
      }
    }
  >
    <ForwardRef(SafeAreaView)>
      <Component
        animationType="slide"
        hardwareAccelerated={false}
        visible={true}
      >
        <Main
          clearAll={[Function]}
          maxRequests={20}
          nRequests={Array []}
          onPress={[Function]}
          onPressClose={[MockFunction]}
          onPressDetail={[Function]}
          reduxActions={Array []}
          rnRequests={Array []}
          testId="mainScreen"
          visible={true}
        />
      </Component>
    </ForwardRef(SafeAreaView)>
  </ContextProvider>
</Provider>
`;
